conf/ports.py

```diff
@@ -81,20 +83,37 @@ def configure_flow_profiles(self, iface):
         if iface == "core":
             self.flow_profiles = [6, 9]
 
-    def init_datapath(self, **kwargs):
+    def configure_cndp(self, jsonc_file, iface, num_workers):
+        self.cndp_jsonc_file = jsonc_file
+        if iface == "access":
+            self.cndp_lport_start_index = 0
+        elif iface == "core":
+            self.cndp_lport_start_index = num_workers
+        else:
+            raise Exception('Unknown interface {}: '.format(iface))
+
+    def init_datapath(self, cndp=False, **kwargs):
         # Initialize PMDPort and RX/TX modules
         name = self.name
-        fast = PMDPort(name="{}Fast".format(name), **kwargs)
+        if not cndp:
+            fast = PMDPort(name="{}Fast".format(name), **kwargs)
+
         self.fpi = Merge(name="{}PortMerge".format(name))
         self.fpo = WorkerSplit(name="{}QSplit".format(name))
 
         for qid in range(self.num_q):
-            fpi = QueueInc(name="{}Q{}FastPI".format(name, qid), port=fast.name, qid=qid)
+            if cndp:
+                lport_index = self.cndp_lport_start_index + qid
+                fast = CndpPort(jsonc_file=self.cndp_jsonc_file, lport_index=lport_index)
```

Comment:
Why not having only these two lines and you can add another line indicating `qid=0` inside the `if cndp` statement? So, `fpi` and `fpo` are generic (same for both DPDK and CNDP: `qid=qid`)